### ISSUE TITLE
globbed paths are always evaluated after mounted paths

### ISSUE THREAD

**Issue by pat** [hanami/router#262]
Continuing on from #261, I've determined why my slice isn't being picked up - because it's a globbed path, and I can see in Hanami::Router#lookup that mounted paths are given priority. So, in the following case:

```ruby
# in config/routes.rb
slice :foo, at: "/foo"
slice :main, at: "/"
```

```ruby
# in slices/foo/config/routes.rb
get "/*path", to: "foo_stuff"
```

```ruby
# in slices/main/config/routes.rb
mount MiniApp, at: "/"
```

`MiniApp` will always return a response - and so, `/foo/*path` will never be given a chance to respond, even though it is namespaced, and is listed ahead of the main slice.

If a mounted path returns a 404 (or 405), should that be considered a final response, or should it be skipped in case there's a globbed path? Or, could we adjust the logic of `lookup` so it respects namespacing and/or route-ordering of globbed paths?

**Comment by pat** [comment]
Wondering if an alternative order of considerations for lookup should be something like:

1. fixed paths
2. paths with variables
3. non-root mounted paths
4. non-root glob paths (i.e. don't start with "/*")
5. root mounted paths
6. root glob paths

The ordering of 3 & 4 and 5 & 6 are interchangeable, and wouldn't stop people from overlapping root routes, but at least would avoid root globs/mounts getting in the way of their non-root equivalents.

Or... could mounts and globs be grouped together (mounting something at `/foo` is kinda equivalent to a glob of `/foo/*path` after all)? Then all that's needed is for root globs/mounts to be listed after the non-root ones, and the rest of the logic should work?

**Comment by timriley** [comment]
This is now fixed by #266.
